### PR TITLE
feat(api): [automation | WhenRegistrationCompletedThenRegisterCurrentEditionCourseUser] implement slice

### LINKED ISSUE #329: [Automation | WhenRegistrationCompletedThenRegisterCurrentEditionCourseUser] Implement slice
**Blockers**
- [ ] https://github.com/CodersCrew/coderscamp/issues/328

**Subtasks**
- [ ] Try to describe subtasks as in previous write slices. Do it on your own! Good luck.

**Event Modeling**
MIRO: https://miro.com/app/board/o9J_lQvnN28=/?moveToWidget=3074457363650368998&cot=14

### PR DESCRIPTION
**Pull request by KrystianKjjk**
Closes #329

### CODE REVIEW

File: packages/api/src/module/write/shared/application/application-command.factory.ts [hunk 1]
```diff
@@ -28,14 +28,17 @@ export class ApplicationCommandFactory {
     const generateId = () => this.idGenerator.generate();
     const currentTime = () => this.timeProvider.currentTime();
 
+    const id = generateId();
```
Comment (nowakprojects) [inline packages/api/src/module/write/shared/application/application-command.factory.ts:31]: What is the purpose of this change? 

Good articles about correlationId and causationId:
https://blog.arkency.com/correlation-id-and-causation-id-in-evented-systems/
https://railseventstore.org/docs/v2/correlation_causation/

**Review by nowakprojects — CHANGES_REQUESTED** [review]

File: packages/api/src/module/write/shared/application/application-command.factory.ts (hunk 1 above)
Comment (KrystianKjjk) [reply packages/api/src/module/write/shared/application/application-command.factory.ts:31]: I did it because of order of generating IDs. 
Before: 
In handler was generated ID, then was generated command Id and then correlationID
Now: 
ID generated in handler is at the end 

so it test lastGeneratedId() returns Id created in handler, not correlationID

**Review by nowakprojects — APPROVED** [review]
